InnerEye/ML/configs/segmentation/HeadAndNeckPaper.py

```diff
@@ -0,0 +1,113 @@
+#  ------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  ------------------------------------------------------------------------------------------
+import logging
+import random
+from typing import Any, Optional
+
+from InnerEye.ML.config import SliceExclusionRule, SummedProbabilityRule, equally_weighted_classes
+from InnerEye.ML.configs.segmentation.HeadAndNeckBase import HeadAndNeckBase
+from InnerEye.ML.utils.model_metadata_util import generate_random_colours_list
+
+# List of structures to segment. The order is important, because different values of num_structures
+# in the constructor will select different prefixes of the list.
+
+STRUCTURE_LIST = ["external", "parotid_l", "parotid_r", "smg_l", "smg_r", "spinal_cord", "brainstem",
+                  "globe_l", "globe_r", "mandible", "spc_muscle", "mpc_muscle", "cochlea_l", "cochlea_r",
+                  "lens_l", "lens_r", "optic_chiasm", "optic_nerve_l", "optic_nerve_r", "pituitary_gland",
+                  "lacrimal_gland_l", "lacrimal_gland_r"]
+RANDOM_COLOUR_GENERATOR = random.Random(0)
+COLOURS = generate_random_colours_list(RANDOM_COLOUR_GENERATOR, len(STRUCTURE_LIST))
+
+
+class HeadAndNeckPaper(HeadAndNeckBase):
+    '''
+    Head and Neck model, as used in the paper.
+    '''
+
+    def __init__(self, num_structures: Optional[int] = None, **kwargs: Any) -> None:
+        '''
+        Creates a new instance of the class.
+        :param num_structures: number of structures from STRUCTURE_LIST to predict (default: all structures)
+        :param kwargs: Additional arguments that will be passed through to the SegmentationModelBase constructor.
+        '''
+        # Number of structures to predict; if positive but less than the length of STRUCTURE_LIST, the relevant prefix
+        # of STRUCTURE_LIST will be predicted.
+        if (num_structures is not None) and \
+                (num_structures <= 0 or num_structures > len(STRUCTURE_LIST)):
+            raise ValueError(f"num structures must be between 0 and {len(STRUCTURE_LIST)}")
+        if num_structures is None:
+            logging.info(f'setting num_structures to: {len(STRUCTURE_LIST)}')
+            num_structures = len(STRUCTURE_LIST)
+        ground_truth_ids = STRUCTURE_LIST[:num_structures]
+        if "ground_truth_ids_display_names" in kwargs:
+            ground_truth_ids_display_names = kwargs.pop("ground_truth_ids_display_names")
+        else:
+            logging.info('using default ground_truth_ids_display_names')
```

Comment:
Here and at other places: Logging strings should use normal capitalization at sentence start.